spec/classes/puppet_agent_config_spec.rb

```diff
@@ -36,25 +36,27 @@
         end
       end
 
-      context 'with runmode => cron' do
-        let :pre_condition do
-          'class { "::puppet": runmode => "cron" }'
-        end
+      unless facts[:osfamily] == 'Archlinux'
+        context 'with runmode => cron' do
```

Comment:
add `unless: facts[:osfamily] == 'Archlinux'` here instead of wrapping the whole block